src/model/mod.rs

```diff
@@ -117,6 +121,64 @@ pub struct CapabilityProperties {
     /// An optional link name to use for this capability
     #[serde(skip_serializing_if = "Option::is_none")]
     pub link_name: Option<String>,
+    /// Optional config to pass to the provider. This can be either a raw string encoded config, or
+    /// a JSON or YAML object
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub config: Option<CapabilityConfig>,
+}
+
+/// Right now providers can technically use any config format they want, although most use JSON.
+/// This enum takes that into account and allows either type of data to be passed
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum CapabilityConfig {
+    Json(serde_json::Value),
+    Opaque(String),
+}
+
+impl<'de> Deserialize<'de> for CapabilityConfig {
+    fn deserialize<D>(deserializer: D) -> Result<CapabilityConfig, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_any(CapabilityConfigVisitor)
+    }
+}
+
+struct CapabilityConfigVisitor;
+
+impl<'de> Visitor<'de> for CapabilityConfigVisitor {
+    type Value = CapabilityConfig;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("string or json")
```

Comment:
Should we clarify here that json == json or YAML?

Reply:
Good idea